### PR TITLE
Use ONNX / Core ML compatible method to broadcast

### PR DESCRIPTION
**Pull request by pcuenca**
Unfortunately `tile` could not be used either, it's still not compatible with ONNX.

This is the method proposed by @harishanand95 in https://github.com/harishanand95/diffusers/commit/8dd4e822f87e1b4259755a2181218797ceecc410.

Addresses the first (easy) part of #284.

### CODE REVIEW

File: src/diffusers/models/unet_2d.py [hunk 1]
```diff
@@ -133,7 +133,7 @@ def forward(
             timesteps = timesteps[None].to(sample.device)
 
         # broadcast to batch dimension
-        timesteps = timesteps.broadcast_to(sample.shape[0])
+        timesteps = timesteps * torch.ones(sample.shape[0], dtype=timesteps.dtype)
```
Comment (patil-suraj) [inline src/diffusers/models/unet_2d.py:136]: Could we add a comment here explaining why we use this to do broadcasting.

**Review by patil-suraj — APPROVED** [review]
LGTM!

**Comment by pcuenca** [comment]
I added the comment and also ensured that the device remains the same.

**Comment by patrickvonplaten** [comment]
Thanks a lot @pcuenca !

**Review by patrickvonplaten — APPROVED** [review]

**Pull request merged**